src/text-field-base.tsx

```diff
@@ -29,35 +30,29 @@ const isValidInputValue = (value?: string, inputType?: React.HTMLInputTypeAttrib
 
 interface FieldEndIconProps {
     Icon: React.FC<IconProps>;
-    className?: string;
+    /** In date fields, we want the icon's background to stay transparent when hovering/pressing it */
+    hasBackgroundColor?: boolean;
```

Comment:
isn't this a design inconsistency?

Reply:
In this field the icon is decorative. The whole field is clickable to show the picker